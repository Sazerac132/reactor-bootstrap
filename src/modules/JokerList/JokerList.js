import React, { Component } from 'react';

import './style.scss';

export default class JokerList extends Component {
  constructor(props) {
    super(props);

    this.state = {
      newTaskName: 'Plot evil deeds'
    };

    this.addTask = this.addTask.bind(this);
    this.changeInputValue = this.changeInputValue.bind(this);
  }

  render() {
    const { addTask } = this;
    const { tasks = [], removeTask } = this.props;
    const { newTaskName } = this.state;

    return (
      <div>
        <input className='input' value={newTaskName} onChange={this.changeInputValue} />
        <button onClick={() => addTask(newTaskName)}>New task</button>
        <ul>
          {tasks.length === 0
            ? <li>No tasks.</li>
            : tasks.map(task => {
              const { name, id } = task;
              return (
                <li key={id}>
                  {name} ...
                  <button onClick={() => removeTask(id)}>
                    Remove
                  </button>
                </li>
              );
            })}
        </ul>
      </div>
    );
  }

  addTask() {
    const { changeInputValue } = this;
    const { addTask } = this.props;
    const { newTaskName } = this.state;

    if (!newTaskName) return;

    addTask(newTaskName);
    changeInputValue('');
  }

  changeInputValue(e) {
    this.setState({
      newTaskName: (typeof e === 'string') ? e : e.target.value
    });
  }
}
