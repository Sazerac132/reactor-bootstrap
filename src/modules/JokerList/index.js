import JokerList from './JokerList';

import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

import {
  addTask,
  removeTask
} from '../../actions/tasks';

const mapStateToProps = (state) => ({
  tasks: state.tasks
});

const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    addTask: (task) => addTask(task),
    removeTask: (id) => removeTask(id)
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(JokerList);
