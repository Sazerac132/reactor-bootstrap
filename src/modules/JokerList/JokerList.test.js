import React from 'react';
import renderer from 'react-test-renderer';

import JokerList from './JokerList';

it('renders correctly', () => {
  const tree = renderer
    .create(<JokerList />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});