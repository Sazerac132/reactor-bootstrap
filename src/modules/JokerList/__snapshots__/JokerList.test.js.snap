// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <input
    className="input"
    onChange={[Function]}
    value="Plot evil deeds"
  />
  <button
    onClick={[Function]}
  >
    New task
  </button>
  <ul>
    <li>
      No tasks.
    </li>
  </ul>
</div>
`;
